opm/output/eclipse/RestartIO.cpp

```diff
@@ -509,8 +541,10 @@ void save(const std::string& filename,
           data::Wells wells,
           const EclipseState& es,
           const EclipseGrid& grid,
+          std::map<std::string, std::vector<double>> extra_data,
```

Comment:
Ok, I think I see your point, but in the asynchronous case we store all variables inside the functor that actually makes the write call so I think it should not actually matter here though.